go/worker/keymanager/worker.go

```diff
@@ -345,6 +349,112 @@ func extractMessageResponsePayload(raw []byte) ([]byte, error) {
 	return cbor.Marshal(msg.Response.Body.Success), nil
 }
 
+func (w *Worker) startClientRuntimeWatcher(rt *registry.Runtime, status *api.Status) error {
+	runtimeID := w.runtime.ID()
+	if status == nil || !status.IsInitialized {
+		return nil
+	}
+	if rt.Kind != registry.KindCompute || rt.KeyManager == nil || !rt.KeyManager.Equal(&runtimeID) {
+		return nil
+	}
+	if w.clientRuntimes[rt.ID] != nil {
+		return nil
+	}
+	w.logger.Info("seen new runtime using us as a key manager",
+		"runtime_id", rt.ID,
+	)
+
+	// Check policy document if runtime is allowed to query any of the
+	// key manager enclaves.
+	var found bool
+	switch {
+	case !status.IsSecure && status.Policy == nil:
+		// Insecure test keymanagers can be without a policy.
+		found = true
+	case status.Policy != nil:
+		for _, enc := range status.Policy.Policy.Enclaves {
+			if _, ok := enc.MayQuery[rt.ID]; ok {
+				found = true
+				break
+			}
+		}
+	}
+	if !found {
+		w.logger.Debug("runtime not found in keymanager policy, skipping",
```

Comment:
Maybe log as a warning or move above "seen new runtime ..." info after the policy check.

Reply:
Yeah makes sense, changed this one into a warning.